### DISCUSSION TITLE
How to show the source in the completion menu?

### DISCUSSION THREAD

**Discussion by OneOfOne** [saghen/blink.cmp#329] [Q&A] [answered]
Like:
![Screenshot_20241113_153207](https://github.com/user-attachments/assets/00daf4a7-8a0d-495c-9ce1-430b8ec81073)

**Comment by OneOfOne** [comment]
```lua
	{
		'saghen/blink.cmp',
		-- enabled = false,
		---@module 'blink.cmp'
		---@type blink.cmp.Config
		opts = {
			windows = {
				autocomplete = {
					draw = {
						columns = { { 'label', 'label_description', gap = 1 }, { 'kind_icon', 'kind' } },
					},
				},
			},
		},
	},

```

returns:

```lua
Error executing vim.schedule lua callback: ...im/lazy/blink.cmp/lua/blink/cmp/windows/autocomplete.lua:366: Invalid autocomplete window draw config
stack traceback:
	[C]: in function 'error'
	...im/lazy/blink.cmp/lua/blink/cmp/windows/autocomplete.lua:366: in function 'get_draw_fn'
	...im/lazy/blink.cmp/lua/blink/cmp/windows/autocomplete.lua:320: in function 'draw'
	...im/lazy/blink.cmp/lua/blink/cmp/windows/autocomplete.lua:139: in function 'open_with_items'
	.../.local/share/nvim/lazy/blink.cmp/lua/blink/cmp/init.lua:57: in function <.../.local/share/nvim/lazy/blink.cmp/
```

**Comment by gegenschall** [comment, marked as answer]
You're probably using the latest tagged version of `blink.cmp` (0.5.1) which seems to have [a different configuration API for `draw`](https://github.com/Saghen/blink.cmp/blob/75cadbcd2657ed01326ca2b0e5e4d78a77127ca3/README.md?plain=1#L406). 

If you want more control over rendering in that version you'd have to use a function, not a table. I believe the code that handles this is [here](https://github.com/Saghen/blink.cmp/blob/75cadbcd2657ed01326ca2b0e5e4d78a77127ca3/lua/blink/cmp/windows/autocomplete.lua#L319C1-L319C9). Alternatively, you could use the latest version of `blink.cmp` from `main`.
